### PR TITLE
OPSEXP-2188: narrow the usage of existingConfigMap for better integration

### PR DESCRIPTION
**Pull request by alxgomz**
Ref: OPSEXP-2188

mostly an RFC i particular regarding the structure of the charts values

### CODE REVIEW

File: charts/alfresco-share/templates/deployment-share.yaml [hunk 1]
```diff
@@ -50,17 +50,31 @@ spec:
...
+            - name: REPO_HOST
+              valueFrom:
+                configmapKeyRef:
+                  name: {{ $repo_details_cm }}
+                  key: {{ .Values.repository.existingConfigMap.keys.host | default "REPO_HOST" }}
+            - name: REPO_PORT
+              valueFrom:
+                configmapKeyRef:
+                  name: {{ $repo_details_cm }}
+                  key: {{ .Values.repository.existingConfigMap.keys.port | default "REPO_HOST" }}
```
Comment (gionn) [inline charts/alfresco-share/templates/deployment-share.yaml:73]:
```suggestion
                  key: {{ .Values.repository.existingConfigMap.keys.port | default "REPO_PORT" }}
```

File: charts/alfresco-share/values.yaml [hunk 2]
```diff
@@ -7,13 +7,14 @@ repository:
...
-  #   CSRF_FILTER_ORIGIN
-  #   EXTERNAL_HOST
   existingConfigMap:
+    # -- a pre-existing configmap which provides expected configuration for Share
+    name:
+    keys:
+      # -- name of the key in the configMap where to find the repository service host
+      host:
+      # -- name of the key in the configMap where to find the repository service port
+      port:
```
Comment (gionn) [inline charts/alfresco-share/values.yaml:17]: I prefer have the defaults here rather than in the template as fallback

**Review by gionn — CHANGES_REQUESTED** [review]

**Review by slohe1 — APPROVED** [review]
LGTM

File: charts/alfresco-sync-service/Chart.lock [hunk 3]
```diff
@@ -10,6 +10,6 @@ dependencies:
   version: 3.1.0
 - name: postgresql
   repository: oci://registry-1.docker.io/bitnamicharts
-  version: 12.5.6
-digest: sha256:e39405a7071d52fabbfb5c13dadabdd196cdfec40637bc26fd2444700c67d7e5
-generated: "2023-06-01T10:44:55.970183+02:00"
+  version: 12.6.6
+digest: sha256:2e35331d0c807c80675a67badca0eeaaef31538151957eabc4dfa904175049cf
+generated: "2023-07-18T13:01:41.970559642Z"
```
Comment (gionn) [inline charts/alfresco-sync-service/Chart.lock:15]: please rebase or amend&pushforce or add an empty commit - these are untested

we probably need to move to alfresco-build PAT

**Review by gionn — CHANGES_REQUESTED** [review]

**Pull request merged**